src/python/pants/init/target_roots_calculator.py

```diff
@@ -68,3 +142,70 @@ def create(cls, options, build_root=None, change_calculator=None):
       return TargetRoots(tuple(SingleAddress(a.spec_path, a.target_name) for a in changed_addresses))
 
     return TargetRoots(spec_roots)
+
+
+
+class EngineChangeCalculator(object):
```

Comment:
I'm not crazy about merging these two classes into the same file - it doesn't feel logically connected to me and feels like it's muddying the water a bit with the combined imports etc.

so at a minimum, I think `EngineChangeCalculator` and friends should go into its own separate file.

but I'm not fully convinced that `pants.init.*` would be any better of an namespace for it other than just being nearer by namespace to `TargetRootsCalculator`? the reason `*ChangeCalculator` lives in `pants.scm.*` is because it uses an SCM under the hood (git) to detect changes and is backed by other code in `pants.scm.*`. imho, it would be just as easy to import and use that here than to move it.

wdyt?

Reply:
gotcha. grouping them on that basis sounds good to me, but I'm still leaning toward separate files per `Calculator` impl to keep things tidy (~"one class per file")?

would sticking them in a nested package (e.g. `src/python/pants/init/roots/*_calculator.py` et al -> e.g. `from pants.init.roots.change_calculator import ChangeCalculator`) suffice as the grouping? or is that too granular?

Reply:
> (~"one class per file")?

Hm... it already wasn't "one class per file" in that `_DependentGraph` is in the same file, and used by both implementations. Not sure I agree about splitting them out unless we have a better strategy for that.

Reply:
that's where the "~" comes in ;)

but sure, whatever you think.